SimG4CMS/Calo/src/CaloSD.cc

```diff
@@ -568,6 +568,7 @@ unsigned int CaloSD::findBoundaryCrossingParent(const G4Track* track, bool markA
 #endif
     boundaryCrossingParentMap_[id] = id;
     trkInfo->setStoreTrack();
+    trkInfo->setIdLastStoredAncestor(id);
```

Comment:
@fabiocos , why we add parameters in CaloSD and not in Phase2SteppingAction? To me it is not normal.

Reply:
@civanch because parameters are added in all the same places where the ```setStoreTrack()``` method is called. If a general update of the whole history management structure is made, of course the definition of ```setIdLastStoredAncestor(id)``` has to be adapted accordingly. But a general refactoring of the whole MC history goes definitely beyond the scope of this PR, whose purpose is to simply extend the existing code to track the latest bit in a shower history persistently stored.

Reply:
if you have already a working proposal for such a refactoring, we may insert the new mechanism there of course...